### ISSUE TITLE
In qtranxf_generateLanguageSelectCode add another option to $type argument

### ISSUE THREAD

**Issue by HowToSolutions** [qTranslate-Team/qtranslate-x#259]
I'm using direct call to qtranxf_generateLanguageSelectCode($type,$id) in my template but since my flags are rendered using CSS, I really don't need anything rendered. Unfortunately, there are only
 'image', 'text', 'both' and 'dropdown' available for $type argument.

At the moment I solved this by using  'image' option and then hiding those images with CSS, but it would really be great, if there would be some sort of 'none' option.

**Comment by johnclause** [comment]
If you do not call `qtranxf_generateLanguageSelectCode` at all, wouldn't it be the same as 'none' option?

**Comment by johnclause** [comment]
BTW, I never researched, but was always curious, why the original author used java script to generate 'select' item for 'dropdown' option? Do you see why? BTW, set of arguments for this function never made sense to me ... it probably needs to be re-designed.

**Comment by HowToSolutions** [comment]
I apologize, I should have been more precise.

I call `qtranxf_generateLanguageSelectCode`, so that the HTML code for Language selection is rendered, I just don't need anything inside it, since I use CSS to render appropriate flags.

I will try to submit a pull request for it.

**Comment by johnclause** [comment]
Thank you, that was quick. Will be released in 3.4.5.

**Comment by johnclause** [comment]
I updated the widget code with 'css_only' option too, the latest https://github.com/qTranslate-Team/qtranslate-x has it all. I would appreciate if you could test it all.

**Comment by HowToSolutions** [comment]
I tested it and it seems fine. For all the types except 'css_only' I compared the generated HTML code of the latest version with the version currently on WordPress and the generated HTML code is the same.

Looking at the generated code, I did notice something about the 'dropdown' type. It uses `onchange` event, so it needs JavaScript to function. In cases, when JavaScript is disabled, the 'dropdown' option will simply not be created as it uses JavaScript for rendering and then it falls back to text type.
